lib/ast/definition.cpp

```diff
@@ -118,9 +123,29 @@ SubsortMap kore_definition::get_subsorts() const {
     }
   }
 
+  subsorts_ = std::optional<SubsortMap>(SubsortMap(subsorts));
   return transitive_closure(subsorts);
 }
 
+SubsortMap kore_definition::get_supersort() {
+
+  if (subsorts_inverted_.has_value()) {
+    return *subsorts_inverted_;
+  }
+
+  auto inverted_subsorts = SubsortMap{};
+
+  for (auto const &[inner, outers] : get_subsorts()) {
+    for (auto *outer : outers) {
+      inverted_subsorts[outer].insert(inner);
+    }
+  }
+
+  subsorts_inverted_ = std::optional<SubsortMap>(
+      SubsortMap(transitive_closure(inverted_subsorts)));
```

Comment:
See related suggestions for renaming this; you also don't need this pile of constructors to assign to a `std::optional`.
```suggestion
  subsorts_inverted_ = transitive_closure(inverted_subsorts);
```